webapp/channels/src/components/admin_console/license_settings/trial_banner/trial_banner.tsx

```diff
@@ -93,14 +89,12 @@ const TrialBanner = ({
     let gettingTrialErrorMsg;
 
     const {formatMessage} = useIntl();
-    const state = store.getState();
-    const getCategory = makeGetCategory();
-    const preferences = getCategory(state, Preferences.UNIQUE);
-    const restartedAfterUpgradePrefValue = preferences.find((pref: PreferenceType) => pref.name === Unique.REQUEST_TRIAL_AFTER_SERVER_UPGRADE);
-    const clickedUpgradeAndStartTrialBtn = preferences.find((pref: PreferenceType) => pref.name === Unique.CLICKED_UPGRADE_AND_TRIAL_BTN);
 
-    const restartedAfterUpgradePrefs = restartedAfterUpgradePrefValue?.value === 'true';
-    const clickedUpgradeAndTrialBtn = clickedUpgradeAndStartTrialBtn?.value === 'true';
+    const restartedAfterUpgradePrefValue = useSelector<GlobalState>((state) => getPreference(state, Preferences.UNIQUE, Unique.REQUEST_TRIAL_AFTER_SERVER_UPGRADE));
+    const clickedUpgradeAndStartTrialBtn = useSelector<GlobalState>((state) => getPreference(state, Preferences.UNIQUE, Unique.CLICKED_UPGRADE_AND_TRIAL_BTN));
+
+    const restartedAfterUpgradePrefs = restartedAfterUpgradePrefValue === 'true';
+    const clickedUpgradeAndTrialBtn = clickedUpgradeAndStartTrialBtn === 'true';
```

Comment:
0/5 You can add this check to the selector:

```suggestion
    const restartedAfterUpgradePrefValue = useSelector<GlobalState>((state) => getPreference(state, Preferences.UNIQUE, Unique.REQUEST_TRIAL_AFTER_SERVER_UPGRADE) === 'true');
    const clickedUpgradeAndStartTrialBtn = useSelector<GlobalState>((state) => getPreference(state, Preferences.UNIQUE, Unique.CLICKED_UPGRADE_AND_TRIAL_BTN) === 'true');
```

It makes it a bit less boilerplaty, but 0/5 on other possible implications.

Reply:
I can actually do a bit better than that by using the `getBool` preference selector. I'm not sure why I didn't do that previously